dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java

```diff
@@ -55,34 +60,52 @@ public Object put(Object id, Object item, String keyspace) {
 
     @Override
     public boolean contains(Object id, String keyspace) {
-        return (get(id, keyspace) != null);
-
+        return get(id, keyspace) != null;
     }
 
     @Override
     public Object get(Object id, String keyspace) {
-        return daprClient.getState(statestoreName, keyspace + "-" + id.toString(), Object.class).block().getValue();
+        String key = resolveKey(keyspace, id);
+
+        return daprClient.getState(stateStoreName, key, Object.class).block().getValue();
     }
 
     @Override
     public <T> T get(Object id, String keyspace, Class<T> type) {
-        Map<String, String> meta = Map.of("contentType", "application/json");
-        GetStateRequest stateRequest = new GetStateRequest(statestoreName, keyspace + "-" + id.toString())
-                .setMetadata(meta);
+        String key = resolveKey(keyspace, id);
+        GetStateRequest stateRequest = new GetStateRequest(stateStoreName, key).setMetadata(CONTENT_TYPE_META);
+
         return daprClient.getState(stateRequest, TypeRef.get(type)).block().getValue();
     }
 
     @Override
     public Object delete(Object id, String keyspace) {
-        daprClient.deleteState(statestoreName, keyspace + "-" + id.toString()).block();
-        return null;
+        var result = get(id, keyspace);
```

Comment:
@artur-ciocanu we need this to return the object right?

Reply:
Yes this is according to the KeyValueAdapter contract described here: https://github.com/spring-projects/spring-data-keyvalue/blob/main/src/main/java/org/springframework/data/keyvalue/core/KeyValueAdapter.java#L83